website/content/en/docs/reference/configuration/unit-tests.md

```diff
@@ -0,0 +1,384 @@
+
+```bash
+vector test /etc/vector/*.toml
+```
+
+Specifying multiple files is useful if you want to, for example, keep your unit tests in a separate
+file from your pipeline configuration. Vector always treats multiple files as a single, unified
+configuration.
+
+## Verifying output {#verifying}
+
+You can use [VRL assertions][assertions] to verify that the output of the transform(s) being tested
+conforms to your expectations. VRL provides two assertion functions:
+
+* [`assert`][assert] takes a [Boolean expression][boolean] as its first argument. If the Boolean
+  resolves to `false`, the test fails and Vector logs an error.
+* [`assert_eq`][assert_eq] takes any two values as its first two arguments. If those two values
+  aren't equal, the test fails and Vector logs an error.
+
+With both functions, you can supply a custom log message to be emitted if the assertion fails:
+
+```ruby
+# Named argument
+assert!(1 == 2, message: "the rules of arithmetic have been violated")
+assert_eq!(1, 2, message: "the rules of arithmetic have been violated")
+
+# Positional arguments are also valid
+assert!(1 == 2, "the rules of arithmetic have been violated")
+assert_eq!(1, 2, "the rules of arithmetic have been violated")
+```
+
+{{< info title="Make your assertions infallible" >}}
+We recommend making `assert` and `assert_eq` invocations in unit tests [infallible] by applying the
+bang (`!`) syntax, as in `assert!(1 == 1)` rather than `assert(1 == 1)`. The `!` indicates that the
+VRL program should abort if the condition fails.
+
+[infallible]: /docs/reference/vrl/#fallibility
+{{< /info >}}
+
+If you use the `assert` function, you need to pass a [Boolean expression][boolean] to the function
+as the first argument. Especially useful when writing Boolean expressions are the [type
+functions][type], functions like [`exists`][exists], [`includes`][includes],
+[`is_nullish`][is_nullish] and [`contains`][contains], and VRL [comparisons]. Here's an example
+usage of a Boolean expression passed to an `assert` function:
+
+```toml
+[[tests.outputs.conditions]]
+type = "vrl"
+source = '''
+assert!(is_string(.message) && is_timestamp(.timestamp) && !exists(.other))
+'''
+```
+
+In this case, the VRL program (under `source`) evaluates to a single Boolean that expresses the
+following:
+
+* The `message` field must be a string
+* The `timestamp` field must be a valid timestamp
+* The `other` field must not exist
+
+It's also possible to break a test up into multiple `assert` or `assert_eq` statements:
+
+```toml
+source = '''
+assert!(exists(.message), "no message field provided")
+assert!(!is_nullish(.message), "message field is an empty string")
+assert!(is_string(.message), "message field has as unexpected type")
+assert_eq!(.message, "success", "message field had an unexpected value")
+assert!(exists(.timestamp), "no timestamp provided")
+assert!(is_timestamp(.timestamp), "timestamp is invalid")
+assert!(!exists(.other), "extraneous other field doesn't belong")
```

Comment:
This line is oddly colored, not sure why 🤔

Reply:
I changed the content to get around that. TOML seems to really not love all these nested quotation marks.